### ISSUE TITLE
Examples missing templating

### ISSUE THREAD

**Issue by coatless** [stdfin/random#1]
FYI, when compiling standalone example in the README I receive:

```bash
error: use of class template 'sitmo::threefry_engine' requires template arguments
    sitmo::threefry_engine eng1, eng2;
```

Error Source: 

https://github.com/stdfin/random/blob/5f7f3eb2c30630cae781d0c95d79aee105624e6f/include/stdfin/random/threefry_engine.hpp#L109

Looks like it needs:

```cpp 
typename UIntType=uint64_t
```

And then an empty template in the examples:

```cpp 
stdfin::threefry_engine<> eng1, eng2;
```

**Comment by stdfin** [comment]
Hi James,

Another solution is to change the example (instead of the template) and have it use the typedef at the bottom of the app:

typedef threefry_engine<uint32_t, 32, 13> threefry; // default

I’m extremely busy right now, but the choice would have to be made by investigating which of the two options is the most C++11 standard compliant (first) and most convenient for users (second). Any ideas?

Thanks,
Thijs

**Comment by rstub** [comment]
I would go with using `threefry` in the example. 

Even though, it might still make sense to specify a default for `UIntType`. Given that the default for `w` is 32, this probably should be `uint32_t`:

https://github.com/stdfin/random/blob/5f7f3eb2c30630cae781d0c95d79aee105624e6f/include/stdfin/random/threefry_engine.hpp#L109-L112

But why is the default for `r` here 20 while it is 13 with the `typedef`?

**Comment by stdfin** [comment]
I agree. I’ll change the example to use the threefry typedef. Also change types indeed to std::unint32_t.

The round 20 and 13 are based on two different sources:
* 20 is from the original paper this engine is based on. 
* 13 is the lower bound of rounds (computation cost) for which the engine still passes all the test in TestU01https://en.wikipedia.org/wiki/TestU01 <testu01https://en.wikipedia.org/wiki/TestU01>

I think it’s safest to set things to 20.
